### ISSUE TITLE
[Bug] outcome_transform ignored for ModelListGP.fantasize

### ISSUE THREAD

**Issue by JackBuck** [pytorch/botorch#1333]
# 🐛 Bug
The `outcome_transform` is not applied when using `ModelListGP.fantasize` meaning that the scale of the fantasy models can be magnified/shrunk.

## To reproduce

**Code snippet to reproduce**
```python
import torch
from botorch.models import ModelListGP, SingleTaskGP
from botorch.models.transforms import Standardize
from botorch.utils.sampling import manual_seed
from botorch.sampling import SobolQMCNormalSampler


# Training data
with manual_seed(1234):
    X = torch.rand(20, 1)
    Y = -100*X

# GP model
models_1d = [
    SingleTaskGP(X, Y[:, i].unsqueeze(-1), outcome_transform=Standardize(m=1))
    for i in range(Y.shape[-1])
]
model = ModelListGP(*models_1d)
model.eval()

# Fantasy models
target_x = torch.tensor([[0.5]])
sampler = SobolQMCNormalSampler(100, seed=9876)
fantasy_models = model.fantasize(target_x, sampler=sampler)

# Print results
print(f"Original model posterior mean: {model.posterior(target_x).mean.item()}")
print(f"Fantasy model posterior mean (average): {fantasy_models.posterior(target_x).mean.mean().item()}")

```

**Output**
We can see in the printed output that the fantasy models have a very different posterior mean.
```
Original model posterior mean: -48.430667877197266
Fantasy model posterior mean (average): -118.550048828125
```
If we remove the `outcome_transform` then the two means are instead very close, which is the expected behaviour.
```
Original model posterior mean: -44.10302734375
Fantasy model posterior mean (average): -44.10468673706055
```

## Expected Behavior
I would expect the `outcome_transform` to be applied in the correct place so that it does not introduce a bias in the fantasy models.

## System information

**Please complete the following information:**
- BoTorch version: 0.6.5
- GPyTorch version: 1.8.0
- PyTorch version: 1.12.0
- Python version: 3.10.5
- Computer OS: Ubuntu 20.04.4 LTS (Focal Fossa)

## Additional context
I believe the problem is in the `condition_on_observations` method. For a `SingleTaskGP`, this method contains the lines:
```python
        if hasattr(self, "outcome_transform"):
            # we need to apply transforms before shifting batch indices around
            Y, noise = self.outcome_transform(Y, noise)
```
However, for `ModelListGP` the transform is not applied and the untransformed fantasy observation is passed straight to `super().get_fantasy_model`.

**Comment by Balandat** [comment]
Thanks for flagging this, this does seem like a bug on our end. We'll work on a fix.

**Comment by JackBuck** [comment]
Thanks!

**Comment by esantorella** [comment]
Closed by #1338
